fmp_server.py

```diff
@@ -38,7 +38,22 @@ def recvAnsConnection(self, packet_id, data, seq):
         connection_data = pati.ConnectionData.unpack(data)
         self.server.debug("Connection: {!r}".format(connection_data))
         self.sendNtcLogin(3, connection_data, seq)
-
+    
+    def recvReqShut(self, packet_id, data, seq):
+        """ReqShut packet.
+
+        ID: 60100100
+        JP: 切断要求
+        TR: Disconnection request
+        """
+        # 1: full logout(?)
+        # 2: logout to different server(?)
+        login_type, = struct.unpack(">B", data)
+        if login_type == 1 and self.session.layer == 0:
```

Comment:
This part shouldn't expose the `Session` class internal logic and should be implemented as a session method instead like `session.shutdown(login_type)`.